test/integration/test.py

```diff
@@ -0,0 +1,178 @@
+import base64
+import json
+import os
+import unittest
+
+import jwt
+import psycopg2
+import requests
+
+
+class IntegrationTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.conn = psycopg2.connect(
+            database=os.environ.get("PGDATABASE", "userdb"),
+            user=os.environ.get("PGUSER", "postgres"),
+            password=os.environ.get("PGPASSWORD", None),
+            host=os.environ.get("PGHOST", None),
+            port=os.environ.get("PGPORT")
+        )
+        token = jwt.encode(
+            {"role": "api-user-create"},
+            key=os.environ.get("JWT_SECRET",
+                "a_test_only_postgrest_jwt_secret"),
+            algorithm="HS256",
+            headers={"alg":"HS256"}
+        )
+        cls.headers = {
+            "Content-Type": "application/json",
+            "Accept": "application/json"
+        }
+        cls.auth_headers = {
+            "Authorization": f"Bearer {token.decode('utf-8')}"
+        }
+        cls.auth_headers.update(cls.headers)
+        cls.cleanDB(cls)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.conn.close()
+
+    def setUp(self):
+        query = """
+            INSERT INTO
+                hosts (name,maxusers)
+                VALUES ('test.hashbang.sh','500');
+        """
+        cur = self.conn.cursor()
+        cur.execute(query)
+        self.conn.commit()
+
+    def tearDown(self):
+        self.cleanDB()
+
+    def cleanDB(self):
+        cur = self.conn.cursor()
+        cur.execute("TRUNCATE passwd, hosts, ssh_public_key, openpgp_public_key CASCADE;")
+        self.conn.commit()
+
+    def test_can_connect_to_api(self):
+        """Can Connect to API Server"""
+        url = "http://userdb-postgrest:3000/"
+        res = requests.get(url)
+        data = res.json()
+        self.assertEqual(data["info"]["title"], "PostgREST API")
+
+
+    def test_cannot_create_user_anonymously(self):
+        """Cannot create user anonymously via PostgREST"""
+        url = "http://userdb-postgrest:3000/passwd"
+        data = {}
+        with open("./keys/id_ed25519.pub", "r") as _f:
+            file_data = _f.read().split('\n')
+            data = {
+                "name": "testuser",
+                "host": "test.hashbang.sh",
+                "data": {
+                    "shell": "/bin/bash",
+                    "ssh_keys": file_data
+                }
+            }
+        res = requests.post(url, json=data, headers=self.headers)
+        self.assertEqual(res.status_code, 401)
+        data = res.json()
+        self.assertIn("permission denied", data.get('message'))
+
+    def test_cannot_create_user_with_invalid_host_and_valid_auth(self):
+        """Can not create user with invalid host and valid auth via PostgREST"""
+        url = "http://userdb-postgrest:3000/passwd"
+        data = {}
```

Comment:
I don't like that `data` contains a thing with field `data`. can you call this `post_body` or similar instead?